server/shared/configFetcher.go

```diff
@@ -54,25 +55,29 @@ func (cf *configFetcher) Fetch() (files []*remote.FileMeta, err error) {
 				// test .woodpecker/ folder
 				// if folder is not supported we will get a "Not implemented" error and continue
 				files, err = cf.remote_.Dir(cf.user, cf.repo, cf.build, ".woodpecker")
-				if err == nil {
+				if err == nil && len(files) != 0 {
 					return filterPipelineFiles(files), nil
 				}
 
 				file, err = cf.remote_.File(cf.user, cf.repo, cf.build, ".woodpecker.yml")
-				if err == nil {
+				if err == nil && len(file) != 0 {
 					return []*remote.FileMeta{{
 						Name: ".woodpecker.yml",
 						Data: file,
 					}}, nil
 				}
 
 				file, err = cf.remote_.File(cf.user, cf.repo, cf.build, ".drone.yml")
-				if err == nil {
+				if err == nil && len(file) != 0 {
 					return []*remote.FileMeta{{
 						Name: ".drone.yml",
 						Data: file,
 					}}, nil
 				}
+
+				if err == nil && len(files) == 0 {
+					return nil, fmt.Errorf("ConfigFetcher: Fallback did not found config")
+				}
```

Comment:
```suggestion
			  return nil, fmt.Errorf("ConfigFetcher: Fallback did not found config")
```

Shouldn't this be enough?

Reply:
well then we would drop the err if there is one

Reply:
if we simplify we otherwise would have to drop retry func

Reply:
> if we simplify we otherwise would have to drop retry func

Upps, we currently don't use the retry anyways as of `return nil, err` in line 83.

Adding some separate tests for empty directories and empty file contents would be nice.